docs/architecture/dapr-for-net-developers/bindings-buildingblock.md

```diff
@@ -0,0 +1,235 @@
+
+The [Dapr documentation for resource bindings][1] provides a complete list of the available bindings and their specific configuration settings.
+
+### Cron binding
+
+Pay close attention to Dapr's Cron binding. It doesn't subscribe to events from an external system. Instead, this binding uses a configurable interval schedule to trigger your application. The binding provides a simple way to implement a background worker to wake up and do some work at a regular interval, without the need to implement an endless loop with a configurable delay. Here's an example of a Cron binding configuration:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: checkOrderBacklog
+  namespace: default
+spec:
+  type: bindings.cron
+  metadata:
+  - name: schedule
+    value: "@every 30m"
+```
+
+In this example, Dapr triggers a service by invoking the `/checkOrderBacklog` endpoint every 30 minutes. There are several patterns available for specifying the `schedule` value. For more information, see the [Cron binding documentation](ocs.dapr.io/operations/components/setup-bindings/supported-bindings/cron/).
+
+## Reference architecture: eShopOnDapr
+
+The accompanying eShopOnDapr reference application implements an output binding example. It triggers the Dapr [SendGrid](https://docs.dapr.io/operations/components/setup-bindings/supported-bindings/sendgrid/) binding to send a user an email when a new order is placed. You can find this binding in the `eshop-email.yaml` file in the components folder:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: sendmail
+  namespace: default
+spec:
+  type: bindings.twilio.sendgrid
+  metadata:
+  - name: apiKey
+    secretKeyRef:
+      name: sendGridAPIKey
+auth:
+  secretStore: eshop-secretstore
+```
+
+This configuration uses the [Twilio SendGrid](https://github.com/dapr/components-contrib/tree/master/bindings/twilio) binding component. Note how the API key for connecting to the service consumes a Dapr secret reference. This approach keeps secrets outside of the configuration file. Read the [Dapr secrets management chapter](secrets-buildingblocks.md) to learn more about Dapr secrets.
+
+The binding configuration specifies a binding component that can be invoked using the `/sendmail` endpoint on the Dapr sidecar. Here's a code snippet in which an email is sent whenever an order is started:
+
+```csharp
+public Task Handle(OrderStartedDomainEvent notification, CancellationToken cancellationToken)
+{
+    var string message = CreateEmailBody(notification);
+    var metadata = new Dictionary<string,string> { 
+      { emailFrom = "eShopOn@dapr.io" },
+      { emailTo = notification.UserName },
+      { subject = $"Your eShopOnDapr order #{notification.Order.Id}" }
+    };
+
+    return _daprClient.InvokeBindingAsync("sendmail", "create", message, metadata, cancellationToken);
+}
```

Comment:
```suggestion
public Task Handle(OrderStartedDomainEvent notification, CancellationToken cancellationToken)
{
    var string message = CreateEmailBody(notification);
    var metadata = new Dictionary<string, string>
    { 
        ["emailFrom"] = "eShopOn@dapr.io",
        ["emailTo"] = notification.UserName,
        ["subject"] = $"Your eShopOnDapr order #{notification.Order.Id}"
    };

    return _daprClient.InvokeBindingAsync("sendmail", "create", message, metadata, cancellationToken);
}
```

Fix compiler error, the collection initializer for a dictionary doesn't work this way - this code does not compile. Also, where does the, `emailFrom`, `emailTo`, and `subject` values comes from? Or are they the string literals for the metadata property names?

Reply:
Yes, the email fields are metadata property names.